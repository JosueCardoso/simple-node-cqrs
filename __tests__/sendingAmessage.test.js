const { commands } = require('../writeModel/domain/commands');
const { InMemoryRepository } = require('../writeModel/infrastructure/inMemory');
const { TestReadModel, TestWriteModel } = require('./testUtils');

process.on('unhandledRejection', reason => {
  throw reason;
});

describe('feature : sending a message', () => {
  describe('given a db containing few users and a chat with id "chat42"', () => {
    describe('when "user42" sends a message in the "chat42"', () => {
      test('then the message should appear in the message list of the "chat42"', async () => {
        expect.assertions(1);
        const {
          saveMessage,
          saveChat,
          getChatOfId,
          getMessageOfId,
          getNextMessageId,
          addOnDbUpdatedListener,
        } = InMemoryRepository({
          chats: {
            chat42: {
              id: 'chat42',
              isClosed: false,
            },
            chat43: {
              id: 'chat43',
              isClosed: false,
            },
          },
          messages: {},
          users: {
            user41: { id: 'user41', username: 'User 41' },
            user42: { id: 'user42', username: 'User 42' },
          },
        });
        const dispatchCommand = TestWriteModel({
          saveMessage,
          saveChat,
          getChatOfId,
          getMessageOfId,
          getNextMessageId,
        });
        TestReadModel({
          addOnDbUpdatedListener,
          viewUpdatedCallback: ({ chatMessages }) => {
            expect(chatMessages).toMatchSnapshot(
              'chat43 should be empty but chat42 should have the message from user42',
            );
          },
        });
        dispatchCommand(
          commands.sendMessage({
            chatId: 'chat42',
            userId: 'user42',
            content: "It's the very first message !",
            sentAt: 1545130856133,
          }),
        );
      });
    });
  });
});
