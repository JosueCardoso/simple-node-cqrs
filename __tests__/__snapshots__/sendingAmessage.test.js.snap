// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature : sending a message given a db containing few users and a chat with id "chat42" when "user42" sends a message in the "chat42" then the message should appear in the message list of the "chat42": chat43 should be empty but chat42 should have the message from user42 1`] = `
Object {
  "chat42": Array [
    Object {
      "content": "It's the very first message !",
      "sentAt": 1545130856133,
      "user": "User 42",
    },
  ],
}
`;
